### PR TITLE
Serializer for device.sua field

### PR DESCRIPTION
**Pull request by denyskravchuk**
Added serialization for the field device.sua

### CODE REVIEW

File: openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java [hunk 1]
```diff
@@ -953,6 +959,63 @@ protected void writeGeoFields(Geo geo, JsonGenerator gen) throws IOException {
     }
   }
 
+  public final void writeSua(UserAgent sua, JsonGenerator gen) throws IOException {
```
Comment (dmytro-babichev) [inline openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java]: I think it's better call it `UserAgent userAgent`. `sua` is the name of the field, and you specified it above

File: openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java [hunk 2]
```diff
@@ -953,6 +959,63 @@ protected void writeGeoFields(Geo geo, JsonGenerator gen) throws IOException {
...
+    gen.writeStartObject();
+    writeSuaFields(sua, gen);
+    writeExtensions(sua, gen);
+    gen.writeEndObject();
+  }
+
+  protected void writeSuaFields(UserAgent sua, JsonGenerator gen) throws IOException {
+    if (sua.getBrowsersCount() != 0) {
+      gen.writeArrayFieldStart("browsers");
+      for (BrandVersion browser : sua.getBrowsersList()) {
```
Comment (dmytro-babichev) [inline openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java]: same here, the name of a field is written above. here you can use `brandVersion` name

File: openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java [hunk 3]
```diff
@@ -953,6 +959,63 @@ protected void writeGeoFields(Geo geo, JsonGenerator gen) throws IOException {
...
+  }
+
+  public final void writeBrowser(BrandVersion browser, JsonGenerator gen) throws IOException {
+    gen.writeStartObject();
+    writeBrandVersionFields(browser, gen);
+    writeExtensions(browser, gen);
+    gen.writeEndObject();
+  }
+
+  public final void writePlatform(BrandVersion platform, JsonGenerator gen) throws IOException {
```
Comment (dmytro-babichev) [inline openrtb-core/src/main/java/com/iabtechlab/openrtb/v2/json/OpenRtbJsonWriter.java]: good example. you have two identical methods, but with different names. this should be avoided

**Review by dmytro-babichev — CHANGES_REQUESTED** [review]
please add a unit test

**Review by dmytro-babichev — CHANGES_REQUESTED** [review]
please add a unit test

**Review by dmytro-babichev — APPROVED** [review]

**Review by simontrasler — APPROVED** [review]
Approved.

**Pull request merged**